### ISSUE TITLE
In-app Messaging campaign's links not triggering app deeplink

### ISSUE THREAD

**Issue by aballano** [firebase/firebase-android-sdk#3181]
### Describe your environment

  * Android Studio version: Android Studio Chipmunk | 2021.2.1 Canary 5
  * Firebase Component: In-App Messaging
  * Component version: `firebase-inappmessaging-display-ktx` with `firebase-bom:28.2.0`

### Describe the problem

* Having an app that handles deeplinks for a specific web domain being `www.example.de`. And with Firebase's In-App Messaging integrated. See (1) below
* When receiving an image or card with a link to such domain `https://www.example.de` and clicking on it in the app when shown
* Then the link will be opened in the default browser instead of being opened by the app itself

The expected behaviour for me here would be that if I have an app representing a website that declares that it handles such website deeplinks then by using Firebase's IAM I should be able to keep my users inside the app enabled for such purpose instead of jumping to the web browser.

Additionally, it doesn't seem possible to prevent the link from opening by overriding its behaviour [as could be interpreted in the docs](https://firebase.google.com/docs/in-app-messaging/customize-messages?platform=android#implement_a_deep_link_handler), mentioning some link handler I could not find anywhere.

#### Relevant Code:

(1)
```
<intent-filter android:autoVerify="true">
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <data
                    android:host="@string/domain"
                    android:scheme="https" />
            </intent-filter>
```

Thank you.

**Comment by argzdev** [comment]
Hi @aballano, thank you for your input. I'll mark this as a feature request so our engineers can have a look and discuss this further once they have the time. Thanks!

**Comment by aballano** [comment]
@argzdev thank you for your fast response. I'm not sure if I see it as a feature but rather a bug as it's affecting Android system's deeplinking feature.

**Comment by argzdev** [comment]
You're right, that sound reasonable. I'll try to repro the case, then get back to you if I need more info or notify an engineer once I can confirm the bug. But I do think I'll be able to confirm it soon since reproducing this does look pretty straight forward.

**Comment by argzdev** [comment]
Hi @aballano, thanks for waiting. After reading the docs, your expectations are indeed what should happen. But as for the issue, I wasn't able to reproduce it. 

Here are the steps I tried:

In Android Studio
1. Creating a clean project with 2 activities: MainActivity, SecondActivity
2. Add artifacts: inappmessaging and dynamic links
3. Add Intent-filters (Similar to your relevant code)

On the Firebase console
1. Create a page link in Dynamic Links section
2. Create an inappmessaging campaign
3. Add page link URL on Button Action
4. Use `Test on device` button

These steps shows me a pop up of the inappmessaging and clicking the button with action will redirect me to SecondActivity without opening a browser link. But it is important to note that the behavior of how an app opens the dynamic link can be manipulated in the settings of the app, see below:

![Screenshot_1638201960](https://user-images.githubusercontent.com/93124282/143908244-0881a3bf-84e1-496d-858a-a74b6541b813.png)

Perhaps you could try something similar by creating a clean project to cross check with your app to see if you're missing anything. If however, you're still encountering the issue, kindly provide me an [MCVE](https://stackoverflow.com/help/minimal-reproducible-example) so I can investigate this further. Thanks!

**Comment by argzdev** [comment]
Hi @aballano, since this issue is stale, I'll close it for now. Feel free to reopen if you have any updates. Thanks!

**Comment by aballano** [comment]
@argzdev Hello, sorry for the late response (was on vacation). The initial issue that I wanted to report didn't had dynamic links actually, reason being that we didn't want to add more libraries, extra code or extra steps for our team to perform. So I would like to ask you:

If the inappmessaging receives a link to an URL that the app already handles vía intent-filter only, shouldn't that suffice for the app to open it?

**Comment by argzdev** [comment]
@aballano, yes that should be sufficient. The expected behavior is that if the intent-filter is properly setup, the app should handle it without redirect its users to a browser (if this is how you set it up on your Firebase console). However in your case, since you've already added `android:autoVerify="true"` in your manifest, I would assume that the issue is the dynamic link setup in the Firebase console where the URL is set.

Could you check the following and see if you're missing anything?

1. Check the page link in your Firebase console _Dynamic Links_ section if it's properly setup. 

<img width="480" alt="dynamic link step 4" src="https://user-images.githubusercontent.com/93124282/145415305-d297ab1a-e1c6-4f7a-aaa6-75330ec3ff61.png">

2. Check your `assetlinks.json` if it is properly setup, you can find the guide [here](https://firebase.google.com/docs/dynamic-links/android/receive#app_links). 

3. You may try using our debugging tool on this [link](https://firebase.google.com/docs/dynamic-links/android/receive), to check the page flowchart of your app.
4. Just a side note, I did mention this in my previous reply, you could check the settings of your app (see my previous message with the image) since the behavior of how an app opens the dynamic link can also be manipulated in the settings.

**Comment by aballano** [comment]
@argzdev I think there's a confusion as I'm not using dynamic links from Firebase but just [regular Android app links](https://developer.android.com/training/app-links/verify-site-associations)

Actually, I just added new short-deeplinks and tested those + the weblinks using `adb shell am start -W -a android.intent.action.VIEW -d <deeplink>` and it works for both `example://article/1234` and `https://www.example.com/article/1234`

I then changed the Firebase campaign to contain the short version and **it did work**! so I believe the problem is more related to the beta version of my app not opening the weblinks rather than firebase. Cannot confirm 100% as I am not able to test campaigns on production right now, but since the short version works I don't see a problem with firebase in that matter.

Thank you for your time and sorry for the noise!
